test/quantization/test_fake_quant.py

```diff
@@ -1,311 +0,0 @@
-import torch
-import torch.cuda
-import torch.jit
-import numpy as np
-from hypothesis import given
-from hypothesis import strategies as st
-import torch.testing._internal.hypothesis_utils as hu
-hu.assert_deadline_disabled()
-from torch.testing._internal.common_utils import TestCase
-from torch.quantization import FakeQuantize
-from torch.quantization import default_observer, default_per_channel_weight_observer
-import io
-import unittest
-
-# Reference method for fake quantize
-def _fake_quantize_per_tensor_affine_reference(X, scale, zero_point, quant_min, quant_max):
-    res = (torch.clamp(torch.round(X * (1.0 / scale) + zero_point), quant_min, quant_max) - zero_point) * scale
-    return res
-
-# Reference method for the gradient of the fake quantize operator
-def _fake_quantize_per_tensor_affine_grad_reference(dY, X, scale, zero_point, quant_min, quant_max):
-    Xq = torch.round(X * (1.0 / scale) + zero_point)
-    mask = (Xq >= quant_min) * (Xq <= quant_max)
-    res = torch.zeros_like(dY)
-    res[mask] = dY[mask]
-    return res
-
-# Helper function used to simulate per-channel fake-quant against any axis
-def _permute_to_axis_zero(X, axis):
-    new_axis_list = list(range(X.dim()))
-    new_axis_list[axis] = 0
-    new_axis_list[0] = axis
-    y = X.permute(tuple(new_axis_list))
-    return y, new_axis_list
-
-# Reference method for fake quantize
-def _fake_quantize_per_channel_affine_reference(X, per_channel_scale, per_channel_zero_point, axis, quant_min, quant_max):
-    X, permute_axis_list = _permute_to_axis_zero(X, axis)
-    res = torch.zeros_like(X)
-
-    for i in range(X.size()[0]):
-        res[i] = (torch.clamp(torch.round(X[i] * (1.0 / per_channel_scale[i]) +
-                  per_channel_zero_point[i]), quant_min, quant_max) - per_channel_zero_point[i]) * per_channel_scale[i]
-
-    out = res.permute(tuple(permute_axis_list))
-    return out
-
-# Reference method for the gradient of the fake quantize operator
-def _fake_quantize_per_channel_affine_grad_reference(dY, X, per_channel_scale, per_channel_zero_point, axis, quant_min, quant_max):
-    X, permute_axis_list = _permute_to_axis_zero(X, axis)
-    Xq = torch.zeros_like(X)
-    for i in range(X.size()[0]):
-        Xq[i] = torch.round(X[i] * (1.0 / per_channel_scale[i]) + per_channel_zero_point[i])
-    Xq = Xq.permute(tuple(permute_axis_list))
-    mask = (Xq >= quant_min) * (Xq <= quant_max)
-    res = torch.zeros_like(dY)
-    res[mask] = dY[mask]
-    return res
-
-def to_tensor(X, device):
-    return torch.tensor(X).to(device=torch.device(device), dtype=torch.float32)
-
-NP_RANDOM_SEED = 19
-tolerance = 1e-6
-
-class TestFakeQuantizePerTensor(TestCase):
-
-    @given(device=st.sampled_from(['cpu', 'cuda'] if torch.cuda.is_available() else ['cpu']),
-           X=hu.tensor(shapes=hu.array_shapes(1, 5,),
-                       qparams=hu.qparams(dtypes=torch.quint8)))
-    def test_forward_per_tensor(self, device, X):
```

Comment:
Some of the fake-quant tests are actually tests of the aten fake-quant op. These op tests should be ideally moved to test_quantized_op.py

Reply:
yeah will add a todo